pinot-query-planner/src/main/java/org/apache/pinot/query/planner/logical/worker/WorkerAssignmentStrategyFactory.java

```diff
@@ -0,0 +1,35 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.query.planner.logical.worker;
+
+import java.util.Map;
+import org.apache.pinot.query.routing.WorkerManager;
+
+
+public class WorkerAssignmentStrategyFactory {
+  private final WorkerManager _workerManager;
+
+  public WorkerAssignmentStrategyFactory(WorkerManager workerManager) {
+    _workerManager = workerManager;
+  }
+
+  public WorkerAssignmentStrategy create(Map<String, String> queryOptions, long requestId) {
+    return new DefaultWorkerAssignmentStrategy(_workerManager, requestId);
+  }
```

Comment:
in order for the SPI to work. the constructor of all implementation of WorkerAssignmentStrategy should take the exact same argument for constructors. 

see. `QuerySchedulerFactor` it has 
```
private static QueryScheduler getQuerySchedulerByClassName(String className, PinotConfiguration schedulerConfig,
      QueryExecutor queryExecutor, ServerMetrics serverMetrics, LongAccumulator latestQueryTime) {
        // ...
    return (QueryScheduler) constructor.newInstance(schedulerConfig, queryExecutor, serverMetrics, latestQueryTime);
  }
```

Reply:
I haven't added the reflection logic yet. Was thinking of adding it in the PR where I add the colocated join strategy. But yes we should implement the SPI in such a way that the dependencies for all implementations should be the same.

On a side-note, do we want to be using reflection in the query hot-path?